spotbugs/src/main/java/edu/umd/cs/findbugs/detect/CollectionIteratorCheck.java

```diff
@@ -0,0 +1,467 @@
+import edu.umd.cs.findbugs.ba.vna.ValueNumber;
+import edu.umd.cs.findbugs.ba.vna.ValueNumberDataflow;
+import edu.umd.cs.findbugs.ba.vna.ValueNumberFrame;
+import edu.umd.cs.findbugs.ba.vna.ValueNumberSourceInfo;
+
+/**
+ * @since ?
+ *
+ */
+public class CollectionIteratorCheck implements Detector {
+
+    private final BugReporter bugReporter;
+    private ClassContext classCtx;
+    /**
+     * @param bugReporter
+     */
+    public CollectionIteratorCheck(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
+
+    @Override
+    public void visitClassContext(ClassContext classContext) {
+        this.classCtx = classContext;
+        Method[] methodList = this.classCtx.getJavaClass().getMethods();// 得到所有的方法列表
+        for (Method method : methodList) {// 遍历方法
+            if (method.getCode() == null) {
+                continue;
+            }
+            if ("<init>".equals(method.getName()) || "<clinit>".equals(method.getName())) {// gouzao class init
+                continue;
+            }
+            CFG cfg;
+            try {
+                cfg = classContext.getCFG(method);// 拿到方法
+                if (null == cfg) {
+                    return;
+                }
+                ConstantPoolGen constPool = classContext.getConstantPoolGen();
+                Collection<Location> locationCollection = cfg.orderedLocations();// 得到方法的指令集
+                ArrayList<Location> locationList = new ArrayList<>();
+                locationList.addAll(locationCollection);
+                analyzeContent(locationList, constPool, classContext, method);// 分析当前方法的所有指令集
+            } catch (MethodUnprofitableException me) {
+                bugReporter.logError("Detector " + this.getClass().getName() + " caught exception", me);
+            } catch (Exception e) {
+                bugReporter.logError("Detector " + this.getClass().getName() + " caught exception", e);
+            }
+        }
+    }
+
+    private void analyzeContent(ArrayList<Location> locationList, ConstantPoolGen constPool, ClassContext classContext,
+            Method method) throws DataflowAnalysisException, CFGBuilderException {
+        for (int i = 0; i < locationList.size(); i++) {// 遍历指令
+            Location location = locationList.get(i);// 得到具体指令
+            InstructionHandle insHandle = location.getHandle();
+            if (insHandle == null) {
+                continue;
+            }
+            Instruction ins = insHandle.getInstruction();
+            if (ins instanceof INVOKEINTERFACE || ins instanceof INVOKEVIRTUAL) {// 如果指令为invokeinterface进入
+                int insIndex = 0;
+                if (ins instanceof INVOKEINTERFACE) {
+                    insIndex = ((INVOKEINTERFACE) ins).getIndex();
+                } else if (ins instanceof INVOKEVIRTUAL) {
+                    insIndex = ((INVOKEVIRTUAL) ins).getIndex();
+                }
+                String classNames = getClassOrMethodName(constPool, insIndex, true);
```

Comment:
Can this variable really contain multiple names? If not, then it would probably be better to call it `className`.